src/utils/cache.ts

```diff
@@ -0,0 +1,35 @@
+import type { Stats } from 'fs'
+
+export type FileCache = Map<string, Promise<string>>
+export type LstatCache = Map<string, Promise<Stats>>
+export type ReaddirCache = Map<string, Promise<string[]>>
+
+interface NFTCache {
+  fileCache: FileCache
+  statCache: Map<string, unknown>
+  symlinkCache: Map<string, unknown>
+  analysisCache: Map<string, unknown>
+}
+
+export interface RuntimeCache {
+  // file content
```

Comment:
I found this comment a bit confusing. I think this would make it clearer?

```suggestion
  // IO operations
```

Reply:
I cleared that up